.github/workflows/test.yml

```diff
@@ -209,7 +209,7 @@ jobs:
           esac
 
       - name: "Save HF models cache"
-        if: ( steps.changed-src-files.outputs.any_changed == 'true' && github.event_name != 'pull_request' && strategy.job-index == 0 )
+        if: ( steps.changed-src-files.outputs.any_changed == 'true' && github.event_name != 'pull_request' )
```

Comment:
We should add the condition `&& steps.cache_restore.outputs.cache-hit != 'true'` here as well